src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReplacementService.cs

```diff
@@ -0,0 +1,350 @@
+using static GovUk.Education.ExploreEducationStatistics.Common.Services.EnumUtil;
+using static GovUk.Education.ExploreEducationStatistics.Data.Model.Services.LocationService;
+using IFootnoteService = GovUk.Education.ExploreEducationStatistics.Data.Model.Services.Interfaces.IFootnoteService;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
+{
+    public class ReplacementService : IReplacementService
+    {
+        private readonly ContentDbContext _contentDbContext;
+        private readonly StatisticsDbContext _statisticsDbContext;
+        private readonly IFilterService _filterService;
+        private readonly IIndicatorService _indicatorService;
+        private readonly ILocationService _locationService;
+        private readonly IFootnoteService _footnoteService;
+        private readonly ITimePeriodService _timePeriodService;
+        private readonly IPersistenceHelper<StatisticsDbContext> _persistenceHelper;
+
+        public ReplacementService(ContentDbContext contentDbContext,
+            StatisticsDbContext statisticsDbContext,
+            IFilterService filterService,
+            IIndicatorService indicatorService,
+            ILocationService locationService,
+            IFootnoteService footnoteService,
+            ITimePeriodService timePeriodService,
+            IPersistenceHelper<StatisticsDbContext> persistenceHelper)
+        {
+            _contentDbContext = contentDbContext;
+            _statisticsDbContext = statisticsDbContext;
+            _filterService = filterService;
+            _indicatorService = indicatorService;
+            _locationService = locationService;
+            _footnoteService = footnoteService;
+            _timePeriodService = timePeriodService;
+            _persistenceHelper = persistenceHelper;
+        }
+
+        public async Task<Either<ActionResult, ReplacementPlanViewModel>> GetReplacementPlan(Guid originalSubjectId,
+            Guid replacementSubjectId)
+        {
+            return await _persistenceHelper
+                .CheckEntityExists<Subject>(originalSubjectId)
+                .OnSuccessDo(() => _persistenceHelper.CheckEntityExists<Subject>(replacementSubjectId))
+                .OnSuccess(() => GetReleaseId(originalSubjectId, replacementSubjectId))
+                .OnSuccess(releaseId =>
+                {
+                    var replacementSubjectMeta = GetReplacementSubjectMeta(replacementSubjectId);
+
+                    var dataBlocks = ValidateDataBlocks(releaseId, originalSubjectId, replacementSubjectMeta);
+                    var footnotes = ValidateFootnotes(releaseId, originalSubjectId, replacementSubjectMeta);
+
+                    return new ReplacementPlanViewModel(dataBlocks, footnotes);
+                });
+        }
+
+        private async Task<Either<ActionResult, Guid>> GetReleaseId(Guid originalSubjectId, Guid replacementSubjectId)
```

Comment:
In the case of amendments the subject could exist against several release "versions" so will this be ok?

Reply:
Good point, I did it like this to avoid an extra Release parameter needing to be passed in and also to validate that somebody isn't doing something silly at an API level like trying to replace a Subject with another across an unrelated Release.

The replacement will always be against the latest version.

If the original is linked in ReleaseSubject to the latest version then I just need to work out what that is.

Reply:
I've made a change now to support getting the latest Release Id from the Subject when it's used in previous release versions.